fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1420,784 @@ def draw_act(self, obj, layout, value):
+        # you've done it!~
+        self.report({"INFO"}, "Success!")
+        return {"FINISHED"}
+
+
+class SM64_CombinedObjectProperties(bpy.types.PropertyGroup):
+    # callbacks must be defined before they are referenced in props
+    def update_preset_behavior(self, context):
+        def update_or_inherit(new_cmd, index, val):
+            if val == "inherit":
+                new_cmd.ineherit_from_export = True
+            else:
+                setattr(new_cmd, f"arg_{index + 1}", val)
+
+        self.behavior_script.clear()
+        bhv_preset = behaviorPresetContents.get(self.preset_behavior_script)
+        if bhv_preset:
+            for cmd in bhv_preset:
+                new_cmd = self.behavior_script.add()
+                new_cmd.macro = cmd[0]
+                [update_or_inherit(new_cmd, j, val) for j, val in enumerate(cmd[1])]
+
+    # internal object used to keep track during exports. Updated by export function
+    context_obj = None
+
+    export_header_type: bpy.props.EnumProperty(
+        name="Header Export",
+        items=[*enumExportHeaderType, ("Custom", "Custom", "No headers are written")],
+        default="Actor",
+    )
+    # level export header
+    level_name: bpy.props.EnumProperty(items=enumLevelNames, name="Level", default="bob")
+    # actor export header
+    group_name: bpy.props.EnumProperty(name="Group Name", default="common0", items=groups_obj_export)
+    # custom export path, no headers written
+    custom_export_path: bpy.props.StringProperty(name="Custom Path", subtype="FILE_PATH")
+
+    # common export opts
+    custom_export_name: bpy.props.StringProperty(name="custom")  # for custom level or custom group
+    model_id: bpy.props.IntProperty(
+        name="Model ID Num", default=0xE2, min=0, description="Export model ID number. A model ID of 0 exports nothing"
+    )
+    object_name: bpy.props.StringProperty(name="Actor Name", default="")
+
+    # collision export options
+    include_children: bpy.props.BoolProperty(
+        name="Include Children",
+        default=True,
+        description="Collision export will include all child objects of linked/selected object",
+    )
+    export_rooms: bpy.props.BoolProperty(
+        name="Export Rooms", description="Collision export will generate rooms.inc.c file"
+    )
+
+    # export options
+    export_bhv: bpy.props.BoolProperty(
+        name="Export Behavior", default=False, description="Export behavior with given object name"
+    )
+    export_col: bpy.props.BoolProperty(
+        name="Export Collision",
+        default=False,
+        description="Export collision for linked or selected mesh that have collision data",
+    )
+    export_gfx: bpy.props.BoolProperty(
+        name="Export Graphics",
+        default=False,
+        description="Export geo layouts for linked or selected mesh that have collision data",
+    )
+    export_script_loads: bpy.props.BoolProperty(
+        name="Export Script Loads",
```

Comment:
is export the right name for this, its more "updating" than an export no?

Reply:
Generally speaking you're appending to an existing script which isn't the same but I think a common understanding of appending with new data is an "export". Also considering it is the nomenclature used for all the other operations done I would say it is good.